data/adapters/encode_enhancer_gene_adapter.py

```diff
@@ -0,0 +1,113 @@
+# chr1	778331	778947	e:chr1:778331-778947	enhancer	NOC2L	ENSG00000188976	959156	Blood.Myeloid.Erythroblast	0.0806398	179809	29.9625	26.7842	8.7749	94.1001400	0.06586	4.07600
+# chr1	778997	779613	e:chr1:778997-779613	enhancer	NOC2L	ENSG00000188976	959156	Blood.Myeloid.Erythroblast	0.0388132	179143	17.7508	7.2314	22.4898	43.6864031	0.06586	4.06500
+# chr1	1157527	1158185	e:chr1:1157527-1158185	enhancer	NOC2L	ENSG00000188976	959156	Blood.Myeloid.Erythroblast	0.0177169	197971	17.8522	6.5309	22.7196	41.6971431	0.03281	2.20300
+# chr1	826563	827090	e:chr1:826563-827090	enhancer	NOC2L	ENSG00000188976	959156	Blood.Myeloid.Erythroblast	0.0122145	131666	18.9075	5.0152	5.9345	25.1611698	0.13036	9.74100
+# #
+# chr1	1024145	1024642	e:chr1:1024145-1024642	enhancer	KLHL17	ENSG00000187961	960684	Blood.Myeloid.Erythroblast	0.0837702	63061	9.9855	7.9918	20.6068	34.0730209	0.10571	2.75800
+# chr1	778331	778947	e:chr1:778331-778947	enhancer	KLHL17	ENSG00000187961	960684	Blood.Myeloid.Erythroblast	0.0608322	181337	29.9625	26.7842	8.7749	94.1001400	0.02866	1.78600
+# chr1	1068807	1069456	e:chr1:1068807-1069456	enhancer	KLHL17	ENSG00000187961	960684	Blood.Myeloid.Erythroblast	0.0507193	107723	3.0948	18.5308	17.4431	28.0899086	0.08053	3.20700
+
+
+class EncodeEnhancerGeneLink(Adapter):
+
+    ALLOWED_LABELS = [
+        'element_gene_link',
+        'regulatory_region',
+    ]
+    ALLOWED_SOURCES = [
+        'ABC',
+        'ENCODE-E2G',
+        'EpiRaction',
+        'graphReg'
+    ]
+
+    def __init__(self, filepath, label, source, source_url, biological_context):
+        if label not in EncodeEnhancerGeneLink.ALLOWED_LABELS:
+            raise ValueError('Ivalid label. Allowed values: ' +
+                             ','.join(EncodeEnhancerGeneLink.ALLOWED_LABELS))
+        if source not in EncodeEnhancerGeneLink.ALLOWED_SOURCES:
+            raise ValueError('Ivalid source. Allowed values: ' +
+                             ','.join(EncodeEnhancerGeneLink.ALLOWED_SOURCES))
+
+        self.filepath = filepath
+        self.dataset = label
+        self.label = label
+        self.source = source
+        self.source_url = source_url
+        self.biological_context = biological_context
+
+        super(EncodeEnhancerGeneLink, self).__init__()
+
+    def process_file(self):
+        with gzip.open(self.filepath, 'rt') as input_file:
+            reader = csv.reader(input_file, delimiter='\t')
+            for row in reader:
+                if row[0].startswith('#'):
+                    continue
+                chr = row[0]
+                start = row[1]
+                end = row[2]
+                class_name = row[4]
+                regulatory_element_id = build_regulatory_region_id(
+                    class_name, chr, start, end)
+                score = row[9]
+
+                if self.label == 'element_gene_link':
```

Comment:
I think we should remove _link both here and from the variant_gene_link.  I know I wrote it that way though.  I haven't written that ticket though so I guess leave it be for now.